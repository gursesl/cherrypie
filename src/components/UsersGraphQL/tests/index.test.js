/* eslint-disable no-unused-vars */
import React from 'react'
import { fromJS } from 'immutable'
import { shallow, mount } from 'enzyme'
// import ApolloClient, { ApolloProvider } from 'react-apollo'
// import gql from 'graphql-tag'
// import { mockServer, makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools'
// import { mockNetworkInterfaceWithSchema } from 'apollo-test-utils'
// import configureStore from 'redux-mock-store'
// import '../../../setupTests'
// import UsersGraphQL, { usersListQuery } from '..'
// import { typeDefs } from '../../../../gqlserver/src/schema'

// const middlewares = []
// const mockStore = configureStore(middlewares)
// const store = mockStore(fromJS({}))

// const myMockServer = mockServer(typeDefs)
// myMockServer.query(usersListQuery)


// const typeDefs = `
//   type User {
//     id: Int
//     name: String
//     email: String
//   }

//   type Query {
//     user: User
//   }
// `

// const schema = makeExecutableSchema({ typeDefs })
// addMockFunctionsToSchema({ schema })

// const mockNetworkInterface = mockNetworkInterfaceWithSchema({ schema })

// const client = new ApolloClient({
//   networkInterface: mockNetworkInterface,
// })

// client.query({
//   query: gql`{ user { id
//     name
//     email } }`,
// })

// const component = (
//   <ApolloProvider store={store} client={client}>
//     <UsersGraphQL />
//   </ApolloProvider>
// )

describe('UserGraphQL component', () => {
  it('Should render correctly', () => {
    // console.log(mount(component))
    // console.log(mount(component).instance().props)
    // console.log(mount(component).instance().props.client.store.getState().apollo.data)
  })
})
