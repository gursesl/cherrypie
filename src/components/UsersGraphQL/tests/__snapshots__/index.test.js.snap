// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGraphQL component Should render error state 1`] = `
<Unknown
  data={
    Object {
      "error": Object {
        "message": "Error Message",
      },
      "loading": false,
      "users": Array [
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "wer@email.com",
          "firstName": "Wera",
          "id": "1",
          "lastName": "Andersen",
          "password": "passw0rd",
          "state": "IL",
          "userName": "weraa",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Smith",
          "id": "2",
          "lastName": "Johnson",
          "password": "passw0rd",
          "state": "IL",
          "userName": "weraa2",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Angle",
          "id": "3",
          "lastName": "Lamar",
          "password": "passw0rd",
          "state": "IL",
          "userName": "angel23",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Wera",
          "id": "4",
          "lastName": "Jeromski",
          "password": "passw0rd",
          "state": "IL",
          "userName": "wera44",
          "userType": "caregiver",
          "zip": "22902",
        },
      ],
    }
  }
/>
`;

exports[`UserGraphQL component Should render loading state 1`] = `
<Unknown
  data={
    Object {
      "error": null,
      "loading": true,
      "users": Array [
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "wer@email.com",
          "firstName": "Wera",
          "id": "1",
          "lastName": "Andersen",
          "password": "passw0rd",
          "state": "IL",
          "userName": "weraa",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Smith",
          "id": "2",
          "lastName": "Johnson",
          "password": "passw0rd",
          "state": "IL",
          "userName": "weraa2",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Angle",
          "id": "3",
          "lastName": "Lamar",
          "password": "passw0rd",
          "state": "IL",
          "userName": "angel23",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Wera",
          "id": "4",
          "lastName": "Jeromski",
          "password": "passw0rd",
          "state": "IL",
          "userName": "wera44",
          "userType": "caregiver",
          "zip": "22902",
        },
      ],
    }
  }
/>
`;

exports[`UserGraphQL component Should render users correctly with data 1`] = `
<Unknown
  data={
    Object {
      "error": null,
      "getUsers": Array [
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "wer@email.com",
          "firstName": "Wera",
          "id": "1",
          "lastName": "Andersen",
          "password": "passw0rd",
          "state": "IL",
          "userName": "weraa",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Smith",
          "id": "2",
          "lastName": "Johnson",
          "password": "passw0rd",
          "state": "IL",
          "userName": "weraa2",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Angle",
          "id": "3",
          "lastName": "Lamar",
          "password": "passw0rd",
          "state": "IL",
          "userName": "angel23",
          "userType": "caregiver",
          "zip": "22902",
        },
        Object {
          "address": "123 Maple St.",
          "address2": "Unit 320",
          "city": "Maperville",
          "email": "weraa@email.com",
          "firstName": "Wera",
          "id": "4",
          "lastName": "Jeromski",
          "password": "passw0rd",
          "state": "IL",
          "userName": "wera44",
          "userType": "caregiver",
          "zip": "22902",
        },
      ],
      "loading": false,
    }
  }
/>
`;

exports[`UserGraphQL component Should render users correctly without data 1`] = `
<Unknown
  data={
    Object {
      "error": null,
      "getUsers": Array [],
      "loading": false,
    }
  }
/>
`;
